src/libmongoc/src/mongoc/mongoc-scram.c

```diff
@@ -1150,4 +1241,222 @@ _mongoc_sasl_prep (const char *in_utf8, int in_utf8_len, bson_error_t *err)
    return bson_strdup (in_utf8);
 #endif
 }
+
+size_t
+_mongoc_utf8_char_length (const char *c)
+{
+   BSON_ASSERT_PARAM (c);
+
+   uint8_t *temp_c = (uint8_t *) c;
+   // UTF-8 characters are either 1, 2, 3, or 4 bytes and the character length
+   // can be determined by the first byte
+   if ((*temp_c & UINT8_C (0x80)) == 0)
+      return 1u;
+   else if ((*temp_c & UINT8_C (0xe0)) == UINT8_C (0xc0))
+      return 2u;
+   else if ((*temp_c & UINT8_C (0xf0)) == UINT8_C (0xe0))
+      return 3u;
+   else if ((*temp_c & UINT8_C (0xf8)) == UINT8_C (0xf0))
+      return 4u;
+   else
+      return 1u;
+}
+
+ssize_t
+_mongoc_utf8_string_length (const char *s)
+{
+   BSON_ASSERT_PARAM (s);
+
+   const uint8_t *c = (uint8_t *) s;
+
+   size_t str_length = 0;
```

Comment:
```suggestion
   ssize_t str_length = 0;
```
Consistent signedness with return value.